57277_fix_missing_set_orientation_api

dbongartz opened the pull request:
SDL & dummy display driver did not set the set_orientation API pointer.
This causes crashes when used.

Added implementation check in display API for set_orientation API.

Fixes: #57277

jfischer-no reviewed (changes requested):
Just

```diff
diff --git a/include/zephyr/drivers/display.h b/include/zephyr/drivers/display.h
index 1b869b30314..92dd2c27ce0 100644
--- a/include/zephyr/drivers/display.h
+++ b/include/zephyr/drivers/display.h
@@ -396,6 +396,7 @@ display_set_pixel_format(const struct device *dev,
  * @param orientation Orientation to be used by display
  *
  * @retval 0 on success else negative errno code.
+ * @retval -ENOSYS If the function is not implemented.
  */
 static inline int display_set_orientation(const struct device *dev,
                                          const enum display_orientation
@@ -404,6 +405,10 @@ static inline int display_set_orientation(const struct device *dev,
        struct display_driver_api *api =
                (struct display_driver_api *)dev->api;
 
+       if (api->set_orientation == NULL) {
+               return -ENOSYS;
+       }
+
        return api->set_orientation(dev, orientation);
 }
```

jfischer-no commented:
> Also works for me. Should I remove the dummy Implementations then?

Please, replace you patches with changes above and force push your branch.

dbongartz commented:
@jfischer-no Should this check then not be added to all or at least most function calls in `display.h`? Feels a bit inconsistent to me to only have it for one function.

jfischer-no commented:
> @jfischer-no Should this check then not be added to all or at least most function calls in `display.h`? Feels a bit inconsistent to me to only have it for one function. Other API's like `led.h` in fact do that.

Perhaps, but to decide what API is mandatory/optional is out of scope of your issue/PR.  Please keep the original intention. Please add errno.h.

```diff
diff --git a/include/zephyr/drivers/display.h b/include/zephyr/drivers/display.h
index 1b869b30314..973c2fa8e79 100644
--- a/include/zephyr/drivers/display.h
+++ b/include/zephyr/drivers/display.h
@@ -20,6 +20,7 @@
  */
 
 #include <zephyr/device.h>
+#include <errno.h>
 #include <stddef.h>
 #include <zephyr/types.h>
```

Commit:
drivers: display: add check for set_orientation API implementation

Some drivers do not implement set_orientation().
To prevent crashes, just error out when not implemented.

Fixes: #57277

Signed-off-by: David Bongartz <david.bongartz@gmail.com>
